Exercise_5.java

```diff
@@ -1,36 +1,65 @@
-class IterativeQuickSort { 
-    void swap(int arr[], int i, int j) 
-    { 
-	//Try swapping without extra variable 
-    } 
-  
+class IterativeQuickSort {
+    void swap(int arr[], int i, int j) {
+        int tmp = arr[i];

```

Comment:
Try to swap without using tmp variable

Reply:
Pushed the changes. Please review the latest commit. Thank you.

Reply:
I have included time and space complexity to all the problems.